docs/DeveloperGuide.adoc

```diff
@@ -96,171 +109,383 @@ The `UI` component,
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+Sorting `Food` objects by lexicographic order is done by the `UniqueFoodList`.
+`Food` objects are able to be sorted because the `Food` class implements the `Comparable<Food>` interface. +
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+How the `UniqueFoodList` works:
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+* Existing `Food` objects in the list are arranged in lexicographic order by `Name`.
+* `UniqueFoodList` sorts the `Food` whenever they are added or edited in the `Model`.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+With this, a top-level idea of the execution of the lexicographical `list` command is given in the sequence diagram below:
 
-image::UndoRedoState0.png[]
+image::ListCommandSequenceDiagram.png[]
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 1: `UI` detects that the user has entered a `String`. +
+Step 2: `UI` passes this `String` into `Logic` to eventually produce a `ListCommand`. +
+Step 3: `Logic` executes the `ListCommand` to filter out relevant `Food` entries from the `Model`. +
+Step 4: All `Food` entries are shown in the GUI. +
+Step 5: `Logic` produces a `CommandResult` which displays its status in the GUI.
 
-image::UndoRedoState1.png[]
+<<<
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+==== Design Considerations
 
-image::UndoRedoState2.png[]
+===== Aspect: How List executes sorting operation
 
-[NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+* **Alternative 1 (current choice):** Sort whenever `Food` is added or edited in the `Model`.
+** Pros: Will guarantee correctness of sorting. Furthermore, since `Food` objects are sorted whenever they are added or edited one by one, each sorting operation will not be too expensive.
```

Comment:
I don't think the cost of sorting works that way, but I'm not maybe I'm not too certain what you mean by "sorting operation will not be too expensive". But it is true that you can insert an element into the sorted array in O(n) time, but not by insert then sort :D